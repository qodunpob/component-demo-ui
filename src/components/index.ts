export * from './gitHubLink'
export * from './preview'
export * from './propList'
export * from './section'
