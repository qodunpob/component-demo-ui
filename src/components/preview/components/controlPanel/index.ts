export * from './ControlPanel'
export * from './ControlPanel.types'
