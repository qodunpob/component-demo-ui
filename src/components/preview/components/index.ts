export * from './controlPanel'
export * from './inspector'
