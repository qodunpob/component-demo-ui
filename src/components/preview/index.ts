export * from './Preview'
export * from './Preview.types'
export type { ControlDefinition } from './components'
