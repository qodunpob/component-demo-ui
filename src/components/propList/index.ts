export * from './PropList'
export * from './PropList.types'
