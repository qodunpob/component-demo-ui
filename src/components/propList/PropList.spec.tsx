import React from 'react'
import { render, type RenderResult } from '@testing-library/react'
import { PropList, type PropListProps } from '~/components'

describe('PropList', () => {
  it('should display provided props and their description', () => {
    const { baseElement } = renderComponent({
      items: [
        { propName: 'firstProp', description: 'description of the firstProp' },
        {
          propName: 'secondProp',
          description: (
            <>
              description of the <i>secondProp</i>
            </>
          )
        }
      ]
    })
    expect(baseElement).toMatchSnapshot()
  })
})

const renderComponent = ({
  items = [],
  ...restProps
}: Partial<PropListProps> = {}): RenderResult =>
  render(<PropList items={items} {...restProps} />)
