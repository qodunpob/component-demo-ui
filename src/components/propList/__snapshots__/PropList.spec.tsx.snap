// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropList should display provided props and their description 1`] = `
<body>
  <div>
    <dl
      class="component-demo-ui-prop-list"
    >
      <dt
        class="component-demo-ui-prop-list-name"
      >
        firstProp
      </dt>
      <dd
        class="component-demo-ui-prop-list-description"
      >
        description of the firstProp
      </dd>
      <dt
        class="component-demo-ui-prop-list-name"
      >
        secondProp
      </dt>
      <dd
        class="component-demo-ui-prop-list-description"
      >
        description of the 
        <i>
          secondProp
        </i>
      </dd>
    </dl>
  </div>
</body>
`;
